pkg/controller/syncsetinstance/syncsetinstance_controller.go

```diff
@@ -264,6 +264,15 @@ func (r *ReconcileSyncSetInstance) Reconcile(request reconcile.Request) (reconci
 		return reconcile.Result{}, err
 	}
 
+	// Prevent users with permissions to create a SyncSet from referencing Secrets in another namespace
+	// and copying them into their clusters. This should be caught in admission, but just incase we have
+	// a check here to refuse to reconcile. Error is logged but not returned as there's no point re-reconciling
+	// this unless the SyncSet is edited. This check is not relevant for SelectorSyncSets which have no namespace.
+	if ssi.Spec.SyncSetRef != nil && r.hasSourceSecretsInAnotherNamespace(ssi, spec) {
+		ssiLog.Warn("syncsetinstance has a source secret in another namespace")
```

Comment:
In case this does get through admission validation, this should bubble up as a condition on the SyncSetInstance. Otherwise, the user will not know why the SyncSet is not applied.

Reply:
Because this is supposed to be handled elsewhere and is just a backup check, it felt like we could get away without carrying code to set a condition similar to some of the other checks above this. Would you agree?

Reply:
I can live with that. Do we know if there are existing syncsets that violate this new validation?

Reply:
Scanned stage, doesn't look like there are any violating.